spec/request/user_request_spec.rb

```diff
@@ -10,55 +10,58 @@
 end
 
 describe '#create', type: :request do
-  context 'when valid' do
-    before do
-      post '/users', params: { user_form: {
+
+  def send_post_request
+    post '/users', params: { user_form: {
         first_name: 'Daniel',
         last_name: 'Martinez',
         role: 'Admin',
         password: 'password',
         email: 'sample@sample.com'
       } }
-    end
+  end
 
-    it { expect(response).to have_http_status(302) }
+  context 'when validate the model user' do
 
-    it 'increase users count' do
-      expect do
-        post '/users', params: { user_form: {
-          first_name: 'Daniel',
-          last_name: 'Martinez',
-          role: 'Admin',
-          password: 'password',
-          email: 'sample@sample.com'
-        } }
-      end.to change { User.count }.by(1)
+    context 'when attributes are present' do
+      it 'does increase users count' do
+        expect do
+          send_post_request
+        end.to change { User.count }.by(1)
+        expect(response).to have_http_status(302)
+      end
     end
-  end
 
-  context 'when not valid' do
-    before do
-      post '/users', params: { user_form: {
-        first_name: 'Daniel',
-        last_name: 'Martinez',
-        role: '',
-        password: '',
-        email: 'sample@sample.com'
-      } }
+    context 'when Role is empty' do
+      it 'does not increase User count' do
+        expect do
+          post '/users', params: { user_form: {
+            first_name: 'Daniel',
+            last_name: 'Martinez',
+            role: '',
+            password: '',
+            email: 'sample@sample.com'
+          } }
```

Comment:
[minor] So as to prevent duplications, we can pass some of params into the method:
```suggestion
          send_post_request(role: '')
```